src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -39,13 +39,361 @@ namespace std
 }
 */
 
-#define NEW_SHADERS
+namespace
+{
+    constexpr const char* VERT_SHADER_NO_UV = 
+R"(#version 430 core
+
+layout (location = 0) in vec3 vPos;
+layout (location = 3) in vec3 vNormal;
+
+layout (location = 0) out vec3 LocalPos;
+layout (location = 1) out vec3 ViewPos;
+layout (location = 2) out vec3 Normal;
+
+layout (set = 1, binding = 0, row_major, std140) uniform UBO {
+    mat4 MVP;
+    mat4 MV;
+    mat3 NormalMat;
+    vec3 EyePos;
+} CamData;
+
+void main()
+{
+    LocalPos = vPos;
+    gl_Position = CamData.MVP*vec4(vPos, 1.0);
+    ViewPos = (CamData.MV*vec4(vPos, 1.0)).xyz;
+    Normal = normalize(CamData.NormalMat*vNormal);
+}
+)";
+    constexpr const char* FRAG_SHADER_NO_UV =
+R"(#version 430 core
+
+layout (location = 0) in vec3 LocalPos;
+layout (location = 1) in vec3 ViewPos;
+layout (location = 2) in vec3 Normal;
+layout (location = 0) out vec4 OutColor;
+
+#define ILLUM_MODEL_MASK 0x0fu
+layout (push_constant) uniform Block {
+    vec3 Ka;
+    vec3 Kd;
+    vec3 Ks;
+    vec3 Ke;
+    vec4 Tf;//w component doesnt matter
+    float Ns;
+    float d;
+    float bm;
+    float Ni;
+    float roughness;
+    float metallic;
+    float sheen;
+    float clearcoatThickness;
+    float clearcoatRoughness;
+    float anisotropy;
+    float anisoRotation;
+    //extra info
+    uint extra;
```

Comment:
factor out into a struct and a built-in header (`irr/builtin/glsl/graphicspipeline/loaders/mtl/common.glsl`)

Reply:
the c++ struct is in SMtl in CMTLPipelineMetadata

Reply:
the c++ struct is in SMtl in CMTLPipelineMetadata
GLSL builtin done in todays commit